extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -161,16 +158,16 @@ export class AmpSidebar extends AMP.BaseElement {
     screenReaderCloseButton.addEventListener('click', () => {
       this.close_();
     });
-    this.element.appendChild(screenReaderCloseButton);
+    element.appendChild(screenReaderCloseButton);
 
     this.registerAction('toggle', this.toggle_.bind(this));
     this.registerAction('open', this.open_.bind(this));
     this.registerAction('close', this.close_.bind(this));
 
-    this.element.addEventListener('click', e => {
+    element.addEventListener('click', e => {
       const target = closestByTag(dev().assertElement(e.target), 'A');
       if (target && target.href) {
-        const tgtLoc = parseUrlDeprecated(target.href);
+        const tgtLoc = Services.urlForDoc(element).parse(target.href);
```

Comment:
should it go to an instance variable? there are 2-3 usages in the component

Reply:
`amp-sidebar` only has one. `amp-iframe` has two, added an instance variable there.